tsinfer/inference.py

```diff
@@ -990,7 +993,7 @@ def __init__(self, sample_data, ancestors_ts, **kwargs):
         self.ancestors_ts_tables = ancestors_ts.dump_tables()
         super().__init__(sample_data, self.ancestors_ts_tables.sites.position, **kwargs)
         self.restore_tree_sequence_builder()
-        self.sample_ids = np.zeros(self.num_samples, dtype=np.int32)
+        self.sample_ids = np.full(self.num_samples, tskit.NULL, dtype=np.int32)
```

Comment:
Wouldn't it be simpler to have ``self.sample_ids = []`` here, and then fill these out below?

Reply:
Yes, I can certainly do this. I was trying to retain the same approach as you had implemented previously, but it's certainly more obvious to create a dynamically sized array and fill it with what you need, rather than mark unused elements with tskit.NULL. Should I re-write using this approach then?

Reply:
I've just tried going here @jeromekelleher , and realise why it's not so easy. The sample_ids is a mapping between the row number in the sample_data file (`0..(num_samples-1)`) and the node ID in the tree sequence builder (which usually have IDs towards then end of the list of nodes, so are much larger numbers).

Instead of `self.sample_ids = []` I could use `self.sample_ids = {}` and fill it with a dictionary mapping row_id->node_id, but since it's usually the case that we are iterating over *all* the samples, that's probably a bit wasteful. In conclusion, I think that my suggestion of filling this with tskit.NULL values at the start, and ignoring any that are <0, is after all, the right one.